src/robusta/core/sinks/opsgenie/opsgenie_sink.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import re
+
+import opsgenie_sdk
+
+from typing import List
+from tabulate import tabulate
+
+from .opsgenie_sink_params import OpsGenieSinkConfigWrapper
+from ...reporting.blocks import (
+    KubernetesDiffBlock,
+    JsonBlock,
+    MarkdownBlock,
+    HeaderBlock,
+    ListBlock,
+    TableBlock,
+    DividerBlock,
+)
+from ...reporting.base import (
+    Finding,
+    Enrichment,
+    FindingSeverity,
+)
+from ..sink_base import SinkBase
+
+
+PRIORITY_MAP = {
+    FindingSeverity.INFO: "P5",
+    FindingSeverity.LOW: "P4",
+    FindingSeverity.MEDIUM: "P3",
+    FindingSeverity.HIGH: "P1"
+}
+
+
+class OpsGenieSink(SinkBase):
+    def __init__(self, sink_config: OpsGenieSinkConfigWrapper, cluster_name: str):
+        super().__init__(sink_config.opsgenie_sink)
+        self.cluster_name = cluster_name
+        self.api_key = sink_config.opsgenie_sink.api_key
+        self.teams = sink_config.opsgenie_sink.teams
+        self.tags = sink_config.opsgenie_sink.tags
+
+        self.conf = opsgenie_sdk.configuration.Configuration()
+        self.conf.api_key['Authorization'] = self.api_key
+
+        self.api_client = opsgenie_sdk.api_client.ApiClient(configuration=self.conf)
+        self.alert_api = opsgenie_sdk.AlertApi(api_client=self.api_client)
+
+    def __eq__(self, other):
```

Comment:
Not related to this PR, but can you at some point document the process of writing a new sink and why we need the __eq__?

It's something that even I would probably forget if I wrote a new sink.

Reply:
I will add documentation for that. 

Regarding the eq, I think it was required at some point, but I went over the code and couldn't find a reason.

Removing it all together